numerics/чебышёв_series_body.hpp

```diff
@@ -0,0 +1,46 @@
+﻿
+#include "numerics/чебышёв_series.hpp"
+
+#include "glog/logging.h"
+
+namespace principia {
+namespace numerics {
+
+template<typename Scalar>
+ЧебышёвSeries<Scalar>::ЧебышёвSeries(std::vector<Scalar> const& coefficients,
+                                     Instant const& t_min,
+                                     Instant const& t_max)
+    : coefficients_(coefficients),
+      degree_(static_cast<int>(coefficients_.size()) - 1),
+      t_min_(t_min),
+      t_max_(t_max) {
+  CHECK_LE(0, degree_) << "Degree must be at least 0";
+  CHECK_LT(t_min_, t_max_) << "Time interval must not be empty";
+  // Precomputed to save operations at the expense of some accuracy loss.
+  Time const duration = t_max_ - t_min_;
+  t_mean_ = t_min_ + 0.5 * duration;
+  two_over_duration_ = 2 / duration;
+}
+
+template<typename Scalar>
+Scalar ЧебышёвSeries<Scalar>::Evaluate(Instant const& t) const {
+  double const scaled_t = (t - t_mean_) * two_over_duration_;
+  double const two_scaled_t = scaled_t + scaled_t;
```

Comment:
I am pretty sure that any decent compiler will hoist this out of the loop, but why depend on that?